state/execution.go

```diff
@@ -179,6 +179,25 @@ func (blockExec *BlockExecutor) ApplyBlock(state State, blockID types.BlockID, b
 	return state, nil
 }
 
+// RevertBlock takes in a reverted block and the new latest block and returns the state
+// to its previous State before reverted block got applied
+// reverted block must be the LastBlockHeight of current state or will return error
+// Will not revert changes to mempool or evpool
+func (blockExec *BlockExecutor) RevertBlock(state State, reverted, newHead *types.Block) (State, error) {
```

Comment:
How should `AppState` get rolled back? Should Tendermint call some ABCI Method to request proxy connection to rollback `AppState`?

Reply:
Great question. Maybe through ABCI we have the app `DeleteVersion` on every version up until the rollback height?